microprofile/tests/junit5/src/main/java/io/helidon/microprofile/tests/junit5/HelidonJunitExtension.java

```diff
@@ -0,0 +1,160 @@
+    @Override
+    public void beforeAll(ExtensionContext context) {
+        Class<?> testClass = context.getRequiredTestClass();
+
+        // prepare configuration
+        Map<String, String> additionalConfig = new HashMap<>();
+        additionalConfig.put("mp.initializer.allow", "true");
+        additionalConfig.put("mp.initializer.no-warn", "true");
+
+        AddConfig[] configAnnotations = testClass.getAnnotationsByType(AddConfig.class);
+
+        for (AddConfig configAnnotation : configAnnotations) {
+            additionalConfig.put(configAnnotation.key(), configAnnotation.value());
+        }
+
+        instance = ConfigProviderResolver.instance();
+        config = instance.getBuilder()
+                .withSources(MpConfigSources.create(additionalConfig))
+                .addDefaultSources()
+                .addDiscoveredSources()
+                .addDiscoveredConverters()
+                .build();
+        instance.registerConfig(config, Thread.currentThread().getContextClassLoader());
+
+        // now let's prepare the CDI bootstrapping
+        SeContainerInitializer initializer = SeContainerInitializer.newInstance();
+
+        HelidonTest testAnnot = testClass.getAnnotation(HelidonTest.class);
+        if (testAnnot != null) {
+            if (!testAnnot.discovery()) {
+                initializer.disableDiscovery();
+            }
+        }
+
+        AddBean[] addBeans = testClass.getAnnotationsByType(AddBean.class);
+        initializer.addExtensions(new AddBeansExtension(testClass, addBeans));
+
+        AddExtension[] addExtensions = testClass.getAnnotationsByType(AddExtension.class);
+        for (AddExtension addExtension : addExtensions) {
+            Class<? extends Extension> extensionClass = addExtension.value();
+            if (Modifier.isPublic(extensionClass.getModifiers())) {
+                initializer.addExtensions(addExtension.value());
+            } else {
+                throw new IllegalArgumentException("Extension classes must be public, but " + extensionClass
+                        .getName() + " is not");
+            }
+        }
+
+        container = initializer.initialize();
+    }
+
+    @Override
+    public void afterAll(ExtensionContext context) {
+        if (container != null) {
+            container.close();
+        }
+        if (instance != null && config != null) {
+            instance.releaseConfig(config);
+        }
+    }
+
+    @Override
+    public <T> T interceptTestClassConstructor(Invocation<T> invocation,
+                                               ReflectiveInvocationContext<Constructor<T>> invocationContext,
+                                               ExtensionContext extensionContext) {
+
+        // we need to replace instantiation with CDI lookup, to properly injection into fields (and constructors)
+        invocation.skip();
+
+        return container.select(invocationContext.getExecutable().getDeclaringClass())
+                .get();
+    }
```

Comment:
Works when injecting to field, but constructor param is not injected in my case:
```java
@HelidonTest(discovery = false)
@AddBeans({
        @AddBean(MessagingLivenessCheck.class),
        @AddBean(TestMessagingBean.class),
})
@AddExtensions({
        @AddExtension(ConfigCdiExtension.class),
        @AddExtension(ServerCdiExtension.class),
        @AddExtension(JaxRsCdiExtension.class),
        @AddExtension(HealthCdiExtension.class),
        @AddExtension(MessagingCdiExtension.class),
})
public class MessagingHealthTest {

    private WebClient client;

    @Inject
    public MessagingHealthTest(ServerCdiExtension server) {
        //ServerCdiExtension server = CDI.current().select(ServerCdiExtension.class).get();
        client = WebClient.builder()
                .baseUri("http://localhost:" + server.port())
                .addReader(JsonpSupport.reader())
                .build();
    }
```
org.junit.jupiter.api.extension.ParameterResolutionException: No ParameterResolver registered for parameter [io.helidon.microprofile.server.ServerCdiExtension arg0] in constructor [public io.helidon.microprofile.messaging.health.MessagingHealthTest(io.helidon.microprofile.server.ServerCdiExtension)].
```

```

Reply:
Turns out `io.helidon.microprofile.tests.junit5.HelidonJunitExtension#interceptTestClassConstructor` didn't get invoked at all. 
But when implementing `org.junit.jupiter.api.extension.ParameterResolver`
```java
    @Override
    public boolean supportsParameter(final ParameterContext parameterContext, final ExtensionContext extensionContext) throws ParameterResolutionException {
        return !container.select(parameterContext.getParameter().getType()).isUnsatisfied();
    }

    @Override
    public Object resolveParameter(final ParameterContext parameterContext, final ExtensionContext extensionContext) throws ParameterResolutionException {
        return container.select(parameterContext.getParameter().getType()).get();
    }
```

It constructor params get resolved

Reply:
I will look into the constructor injection

Reply:
On the other hand people usually do not expect to add constructors to their test classes...

Reply:
But when the test class becomes a bean ... :smile: , there is no other beforeAll invoked after injection

Reply:
More of a nice to have than a must have IMO

Reply:
This seems to be a small "miss-design" in Junit - they obtain argument definitions, argument values, and then their code is ignored, as I use CDI to instantiate the bean (so all of that gets discarded).
I have created a workaround that does not actually select the constructor parameters from CDI and just returns default values (null or primitive type). Added a test as well

Reply:
Thx for the hint `@PostConstruct` is just the beforeAll I needed
```java
@HelidonTest(resetPerTest = false)
@DisableDiscovery
@AddBeans({
        @AddBean(MessagingLivenessCheck.class),
        @AddBean(TestMessagingBean.class),
})
@AddExtensions({
        @AddExtension(ConfigCdiExtension.class),
        @AddExtension(ServerCdiExtension.class),
        @AddExtension(JaxRsCdiExtension.class),
        @AddExtension(HealthCdiExtension.class),
        @AddExtension(MessagingCdiExtension.class),
})
class MessagingHealthTest {

    private WebClient client;

    @Inject
    ServerCdiExtension server;

    @Inject
    TestMessagingBean bean;

    @PostConstruct
    void beforeAll() {
        client = WebClient.builder()
                .baseUri("http://localhost:" + server.port())
                .addReader(JsonpSupport.reader())
                .build();
    }
```